### PR TITLE
appimage_read_file_into_buffer_following_symlinks

### LINKED ISSUE #1: Add appimage_extract_file_following_symlinks that returns file contents
There should be an equivalent of `appimage_extract_file_following_symlinks` that returns the file's contents (as `uchar*` or something) so it can be fed directly into e.g. `QImage` for thumbnail generation without having to save it to a temporary location first.

### PR DESCRIPTION
**Pull request by TheAssassin**
Fixes #1.

### CODE REVIEW

**Comment by TheAssassin** [comment]
Was quite a challenge to get to this point (I had to read into a lot of code, and performed quite some refactoring to make the code more maintainable), but this should do the job.

@kbroulik please review/test.

File: src/libappimage/type1.c [hunk 1]
```diff
@@ -110,13 +110,43 @@ void type1_extract_file(appimage_handler* handler, void* data, const char* targe
...
+        fprintf(stderr, "failed to allocate enough memory for buffer (required: %ul bytes)\n", file_size);
+#endif
+        return false;
+    }
+
+    if (archive_read_data(a, new_buffer, (size_t) file_size) < 0) {
+#ifdef STANDALONE
+        fprintf(stderr, "failed to read data into buffer: %s\n", archive_error_string(a));
+#endif
+        return false;
```
Comment (kbroulik) [inline src/libappimage/type1.c]: Don't you leak the buffer here, given you don't pass it outside for manual `free()` but you also don't dispose of it in here?

Comment (TheAssassin) [reply src/libappimage/type1.c]: Good catch, it must be `free()`d of course.

**Comment by TheAssassin** [comment]
@kbroulik and I tested this code with a lot of AppImages, and it appears to work fine.

**Comment by probonopd** [comment]
Great. Thanks @kbroulik and @TheAssassin

**Pull request merged**